src/ghostfragment/fragmenting/inter_finder.cpp

```diff
@@ -0,0 +1,81 @@
+#include "inter_finder.hpp"
+#include <iostream>
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+
+using namespace ghostfragment::fragmenting;
+
+void find_group_intersections(
+  const system_type& fragments, inter_type inter, std::size_t recent_add,
```

Comment:
`m` should be returned to make this a pure function. AFAIK you're not modifying `log` so it should be passed by `const` reference, same for `group`.

Reply:
That's somewhat why I made the function that creates `log` constant. Would it be better to just pass a constant version into the input?

Reply:
For the `find_group_intersections` would it still be a good idea to just return the map even though it is recursive?

Reply:
Returning from a recursive function is fine, return-value optimization just applies recursively now.